src/v/kafka/server/usage_aggregator.cc

```diff
@@ -108,6 +108,32 @@ auto epoch_time_secs(time_point now) {
       .count();
 }
 
+static ss::lowres_system_clock::time_point round_to_interval(
+  std::chrono::seconds usage_window_width_interval,
+  ss::lowres_system_clock::time_point t) {
+    /// Downstream systems are particularly sensitive to minor issues with
+    /// timestamps not triggering on the configured interval (hours, minutes,
+    /// seconds, etc), this method rounds t to the nearest interval and logs an
+    /// error if this cannot be done within some threshold.
+    const auto interval = usage_window_width_interval;
+    const auto err_threshold = interval < 2min ? interval : 2min;
+    const auto cur_interval_start = t - (t.time_since_epoch() % interval);
+    const auto next_interval_start = cur_interval_start + interval;
+    if (t - cur_interval_start <= err_threshold) {
+        return {cur_interval_start};
+    } else if (next_interval_start - t <= err_threshold) {
+        return {next_interval_start};
+    }
```

Comment:
why are both starts considered? for example, shouldn't exactly one of these two be considered the expected time at which a timer fired, and if it isn't within the error range of that time the error below is logged?